packages/common/src/items/createResource.ts

```diff
@@ -0,0 +1,71 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import { addItemResource } from "@esri/arcgis-rest-portal";
+import HubError from "../HubError";
+import { objectToJsonBlob, stringToBlob } from "../resources";
+import { getPortalApiUrl } from "../urls";
+
+/**
+ * Given an item, and owner, add a resource to the item and returns its url
+ *
+ * @export
+ * @param {string} id
+ * @param {string} owner
+ * @param {*} resource
+ * @param {string} resourceName
+ * @param {IUserRequestOptions} ro
+ * @param {string} [prefix=""]
+ * @return {*}  {Promise<string>}
+ */
+export async function createResource(
+  id: string,
+  owner: string,
+  resource: any,
+  resourceName: string,
+  ro: IUserRequestOptions,
+  prefix: string = ""
+): Promise<string> {
+  try {
+    // Convert resource if needed
+    const extension = resourceName.split(".").pop();
+    const resourceToUpload =
+      // If a json object, convert to blob
+      extension === "json"
```

Comment:
Maybe it's just me, but I find this double ternary a little difficult to parse. I find something like this to be a bit more readable:

```ts
let resourceToUpload = resource;
const extension = resourceName.split(".").pop();

if(extension === "json") {
  resourceToUpload = objectToJsonBlob(resource)
}
if(extension === "txt") {
  resourceToUpload = stringToBlob(resource)
}
```